core/src/main/scala/kafka/server/DelayedElectLeader.scala

```diff
@@ -23,15 +23,16 @@ import org.apache.kafka.common.requests.ApiError
 
 import scala.collection.{Map, mutable}
 
-/** A delayed elect preferred leader operation that can be created by the replica manager and watched
-  * in the elect preferred leader purgatory
+/** A delayed elect leader operation that can be created by the replica manager and watched
+  * in the elect leader purgatory
   */
-class DelayedElectPreferredLeader(delayMs: Long,
-                                  expectedLeaders: Map[TopicPartition, Int],
-                                  results: Map[TopicPartition, ApiError],
-                                  replicaManager: ReplicaManager,
-                                  responseCallback: Map[TopicPartition, ApiError] => Unit)
-    extends DelayedOperation(delayMs) {
+class DelayedElectLeader(
```

Comment:
We don't have to do it here, but one thought I had is if it would be worthwhile to preempt the controller event if we timeout before completion.

Reply:
Yeah. Let's do this in a future PR. I want to revisit how we manager network request/response and the controller event queue.